### PR TITLE
Register and Log In

### PR DESCRIPTION
**Pull request by matthewkohlhaas**
Back end code for the user registration and log in features

### CODE REVIEW

File: app/controllers/users.controller.js [hunk 1]
```diff
@@ -1,41 +1,77 @@
...
+        .exec(function(err, user) {
             if (err) {
-                return next(err);
+                throw err;
+            } else if (!user) {
+                res.status(401).send({successful: false, text: failureMsg});
             } else {
-                req.user = user;
-                next();
+                // check if password matches
```
Comment (ScottSWolfe) [inline app/controllers/users.controller.js]: I'm thinking we don't need the comments in this section. Everything seems pretty self-explanatory. I feel like the comments are more distracting than actually helping to clarify.

File: app/controllers/listings.controller.js [hunk 2]
```diff
@@ -1,22 +1,16 @@
+var verification = require('../utils/verification.util');
 var Listing = require('mongoose').model('Listing');
 
-exports.create = function(req, res, next) {
```
[the remark this answers is not in the document]
Comment (matthewkohlhaas) [reply app/controllers/listings.controller.js]: We don't need it yet, and it did not check for credentials.

File: app/controllers/users.controller.js [hunk 3]
```diff
@@ -1,41 +1,77 @@
...
-    user.save(function(err) {
-        if (err) {
-            return next(err);
-        } else {
-            res.json(user);
-        }
-    });
-};
+const EMAIL_REGEX = /^.+@gatech.edu$/i;
+const MIN_PASSWORD_LENGTH = 6;
```
[the remark this answers is not in the document]
Comment (matthewkohlhaas) [reply app/controllers/users.controller.js]: Yep, I will make that change later today.

File: app/controllers/users.controller.js (hunk 1 above)
Comment (matthewkohlhaas) [reply app/controllers/users.controller.js]: OK. I will remove them if they are not helpful.

**Pull request merged**